src/ruby/yast/logger.rb

```diff
@@ -15,47 +15,52 @@ def y2_logger_helper(level,args)
         args.shift
         caller_frame = 2
       else
-        caller_frame = caller_frame + args.shift
+        caller_frame += args.shift
       end
     end
 
     res = Builtins.sformat(*args)
-    res.gsub!(/%/,"%%") #reescape all %
+    res.gsub!(/%/, "%%") # reescape all %
     caller[caller_frame] =~ /(.+):(\d+):in `([^']+)'/
-    y2_logger(level, "Ruby", $1, $2.to_i, $3, res)
+    y2_logger(level, "Ruby", Regexp.last_match(1), Regexp.last_match(2).to_i, Regexp.last_match(3), res)
 
     if backtrace
       y2_logger_helper(level, [2, "------------- Backtrace begin -------------"])
-      caller(3).each {|frame| y2_logger_helper(level, [4, frame])}
+      caller(3).each { |frame| y2_logger_helper(level, [4, frame]) }
       y2_logger_helper(level, [2, "------------- Backtrace end ---------------"])
     end
   end
 
   module_function
+
```

Comment:
ugh, why??? I really want to have it as tight as possible to y2debug as it modify accessibility of method. maybe even better to use `module_function def y2debug(*args)` ?